### PR TITLE
LibWeb: Implement CSS exponential functions

### PR DESCRIPTION
**Pull request by stelar7**
https://drafts.csswg.org/css-values-4/#exponent-funcs

### CODE REVIEW

**Comment by circl-lastname** [comment]
To not wasate CI resources, a lot your PRs are really doing the same thing - adding CSS math functions. I saw it said that you shouldn't be afraid to make one bigger PR to centralize discussion and speed up checking the source.

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 1]
```diff
@@ -3432,6 +3432,123 @@ ErrorOr<RefPtr<CalculatedStyleValue>> Parser::parse_calculated_value(Vector<Comp
     return CalculatedStyleValue::create(calculation_tree.release_nonnull(), calc_type.release_value());
 }
 
+ErrorOr<NonnullOwnPtr<CalculationNode>> Parser::parse_pow_function(Function const& function)
+{
+    TokenStream stream { function.values() };
+    auto parameters = parse_a_comma_separated_list_of_component_values(stream);
+
+    if (parameters.size() != 2) {
+        return Error::from_string_view("pow() must have exactly two parameter"sv);
```
Comment (konradekk) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:3441]: typo: parameters

File: Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp [hunk 2]
```diff
@@ -390,6 +390,317 @@ ErrorOr<void> InvertCalculationNode::dump(StringBuilder& builder, int indent) co
...
+{
+    auto node_a = m_a->resolve(layout_node, percentage_basis);
+    auto node_a_value = resolve_value(node_a.value(), layout_node);
+
+    auto node_b = m_b->resolve(layout_node, percentage_basis);
+    auto node_b_value = resolve_value(node_b.value(), layout_node);
+
+    auto result = log2(node_a_value) / log2(node_b_value);
+
+    return { Number(Number::Type::Number, result) };
```
Comment (konradekk) [inline Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp:637]: it seems that `b`, the base, is optional and equal to _e_ if not given (maybe thereʼs a `ln` function to use instead in such case?)

**Review by konradekk — COMMENTED** [review]
some remarks

File: Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp (hunk 2 above)
Comment (stelar7) [reply Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp:637]: This is already handled during parsing.

**Comment by AtkinsSJ** [comment]
It's a little bit late to say at this point, but when you have multiple PRs that work together like this, it's a bit less confusing (and involves less rebases) if you either submit them one at a time, or if you mark all but one as drafts. Maybe even actually base each one on the previous. As is, each of these has been trying to insert code in the same place, so every time one gets merged it creates conflicts on all the others.

That is to say, this has conflicts again. :upside_down_face: 

No need to change things, we're nearly done, but in future. And I don't remember if I thanked you before for breaking up the mega-pr of all of these together, so thanks for that!

**Pull request merged**